195 optimize voting

Vitomir2 opened the pull request:
# Description

- create a new condition to check when the price is already updated in the oracle;
- rename the alreadyVotedMapping to hasExecutedForDay in order to generalize and reuse it;
- adapt the variables in the tests and delete redundant mock variable;

# Changes include

- [x] New feature (non-breaking change that adds functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Since there was no point in modifying the contracts and creating the real case scenario, I tested with the false reason that the Price Oracle contract gave me and the flag for an already updated price works fine.

Vitomir2 committed:
optimize voting

create a new mapping to flag when the price is updated already;
create a new function to check if the price is updated already;
put an additional check for the priceUpdatedAlready next with the alreadyVoted;

Vitomir2 committed:
update tests

R-Santev commented on price-oracle/price_oracle.go:
```diff
@@ -25,9 +25,10 @@ import (
-	alreadyVotedMapping = make(map[uint64]bool)
-	calculatedDayNumber = make(map[uint64]uint64)
-	priceVotedEventABI  = contractsapi.PriceOracle.Abi.Events["PriceVoted"]
+	alreadyVotedMapping        = make(map[uint64]bool)
```
Why don't you just rename the alreadyVoted variable to be more common and reuse it for both cases. They do literally the same with priceUpdatedAlready

Vitomir2 committed:
update price updated flag

delete the new mapping for the priceUpdated flags and generalize the alreadyVotedMapping in order to reuse it;
adapt the variables in the tests and delete redundant mock variable;
